### PR TITLE
virtualenv

### PR DESCRIPTION
**Pull request by kalefranz**
This is work that @mingwandroid just pushed to repo.continuum.io/pkgs/free.  The two patches are pretty awesome.

This should resolve https://github.com/conda-forge/staged-recipes/issues/1139

### CODE REVIEW

**Comment by jakirkham** [comment]
Would you mind taking a look @hayd?

File: recipes/virtualenv/build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#!/bin/bash
+
+$PYTHON setup.py install
```
Comment (jakirkham) [inline recipes/virtualenv/build.sh]: Please move this into the `build/script` section and remove `build.sh` and `bld.bat`.

Comment (jakirkham) [inline recipes/virtualenv/build.sh]: Also if this is using `setuptools` to install, this needs `--single-version-externally-managed --record record.txt`.

**Comment by jakirkham** [comment]
What did upstream think about these patches?

**Comment by mingwandroid** [comment]
I am not going to submit them upstream until I am totally happy with them.

I plan to test them by writing python implementations of `ldd` and `otool -L` and then run a battery of tests to verify that they work correctly in every case.

But the fix `virtualenv` on macOS and Linux, so they're at least better than nothing.

**Comment by jakirkham** [comment]
That's completely fair. Am just interested if they will be amenable to these sorts of changes.

**Comment by jakirkham** [comment]
> I plan to test them by writing python implementations of `ldd` and `otool -L`...

Also this would be a really neat tool by itself. Please keep me posted.

File: recipes/virtualenv/build.sh (hunk 1 above)
Comment (kalefranz) [inline recipes/virtualenv/build.sh]: `setuptools` is only used if it's on SYSPATH.  Otherwise `setup.py` falls back to distutils.

Comment (jakirkham) [inline recipes/virtualenv/build.sh]: Figured there was something subtle with `virtualenv` hence the "if this is using `setuptools` to install...". 😄

So you shouldn't need the arguments. The other comment still applies.

File: recipes/virtualenv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  run:
+    - python
+
+test:
+  commands:
+    - virtualenv --help
+    - |
+      virtualenv -p "$(which python)" /tmp/venv-$$                                # [not win]
+      source /tmp/venv-$$/bin/activate                                            # [not win]
+      python -c 'import sys ; sys.exit(not hasattr(sys, "real_prefix"))'          # [not win]
```
Comment (jakirkham) [inline recipes/virtualenv/meta.yaml]: Should we being doing a similar test for Windows? Would it make sense to move this to `run_test.sh`?

File: recipes/virtualenv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+about:
+  home: https://virtualenv.pypa.io/
+  license: MIT
+  license_file: LICENSE.txt
+  summary: Virtual Python Environment builder
+  description: |
+    virtualenv is a tool to create isolated Python environments.
+  doc_url: https://virtualenv.pypa.io/en/latest/
+  dev_url: https://github.com/pypa/virtualenv
+extra:
```
Comment (jakirkham) [inline recipes/virtualenv/meta.yaml]: nit: Please add an empty line above between `about` and `extra`.

File: recipes/virtualenv/meta.yaml (hunk 2 above)
Comment (kalefranz) [inline recipes/virtualenv/meta.yaml]: I'm not sure virtualenv was ever even a problem on windows, and regardless I don't think this PR changes virtualenv's behavior on Windows.

**Comment by kalefranz** [comment]
Ok added the empty line and @jakirkham as a maintainer.  Once this PR is all green I probably won't be swinging back by again.  If there are still problems and someone else wants to take it over, be my guest :)

**Comment by jakirkham** [comment]
No need to make me a maintainer. Please remove me from the list.

**Comment by jakirkham** [comment]
Thanks @kalefranz and @mingwandroid.

**Pull request merged**